internal/security/config/command/proxy/adduser/command.go

```diff
@@ -0,0 +1,340 @@
+	default:
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Associate JWT request failed with code: %d", resp.StatusCode)
+	}
+
+	outputKey := fmt.Sprintf("%s", parsedResponse["key"])
+
+	fmt.Printf("%s\n", outputKey)
+
+	return interfaces.StatusCodeExitNormal, nil
+}
+
+func (c *cmd) ExecuteAddOAuth2() (statusCode int, err error) {
+	if err := c.createConsumer(); err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+	if err := c.addUserToGroup(); err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+
+	// Create an OAuth application
+	// https://docs.konghq.com/hub/kong-inc/oauth2/#create-an-application
+
+	form := url.Values{
+		"name":          []string{c.username}, // use username as application name
+		"redirect_uris": []string{c.redirectUris},
+	}
+
+	// Client ID and client secret are auto-generated if not supplied
+	if len(c.clientID) > 0 {
+		form.Set("client_id", c.clientID)
+	}
+	if len(c.clientID) > 0 {
+		form.Set("client_secret", c.clientSecret)
+	}
+
+	kongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "consumers", c.username, "oauth2"}, "/")
+	c.loggingClient.Info(fmt.Sprintf("creating oauth application at %s", kongURL))
+
+	formVal := form.Encode()
+	req, err := http.NewRequest(http.MethodPost, kongURL, strings.NewReader(formVal))
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Failed to prepare request to create oauth application %s: %w", c.username, err)
+	}
+	req.Header.Add(clients.ContentType, urlEncodedForm)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Failed to send request to create oauth application %s: %w", c.username, err)
+	}
+	defer resp.Body.Close()
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+
+	var parsedResponse map[string]interface{}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+	case http.StatusCreated:
+		if err := json.NewDecoder(bytes.NewReader(responseBody)).Decode(&parsedResponse); err != nil {
+			return interfaces.StatusCodeExitWithError, fmt.Errorf("Unable to parse associate JWT response: %w", err)
+		}
+	case http.StatusConflict:
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Associate JWT request failed (likely due to duplicate ID) with code: %d", resp.StatusCode)
+	default:
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Associate JWT request failed with code: %d", resp.StatusCode)
+	}
+
+	clientID := fmt.Sprintf("%s", parsedResponse["client_id"])
+	clientSecret := fmt.Sprintf("%s", parsedResponse["client_secret"])
+
+	err = json.NewEncoder(os.Stdout).Encode(map[string]interface{}{
+		"client_id":     clientID,
+		"client_secret": clientSecret,
+	})
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Marshaling of client id and secret failed")
+	}
+
```

Comment:
`ExecuteAddJwt` outputs a key to stdout, but this method doesn't, why? Also the README doesn't mention any output from the command(s).

Reply:
Added to documentation "Upon completion, the command outputs a JSON structure containing the client_id and client_secret."  See line 330, the json is serialized to os.Stdout.